src/core/packages/saved-objects/migration-server-internal/src/actions/wait_for_task.ts

```diff
@@ -42,6 +42,19 @@ export interface WaitForTaskCompletionTimeout {
   readonly error?: Error;
 }
 
+/**
+ * When we use `wait_for_completion=false`, we won't get the errors right away, we'll get a
+ * task id. Then we have to query the tasks API with that id and Elasticsearch will tell us
+ * if there was any error in the original task inside a 200 response. In some cases we might
+ * want to retry the original task.
+ */
+export interface WaitForTaskCompletedWithErrorRetryOriginal {
```

Comment:
NIT What about something like `TaskCompletedWithRetriableError` , it is simpler and it already refers to the "Task" instead of the "WaitForTask" action.